### PR TITLE
Add new cases of papr hpt

### PR DESCRIPTION
**Pull request by chloerh**
- set hpt resizing to 'disabled'
- set hpt resizing to 'enabled' and do memory hotplug
- set hpt resizinzg on non-ppc hosts

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Test Result on x86:
-------

 (1/5) type_specific.local.papr_hpt.positive_test.disabled.ppc_only: CANCEL (2.09 s)
 (2/5) type_specific.local.papr_hpt.positive_test.enabled.mem_hotplut.ppc_only: CANCEL (1.94 s)
 (3/5) type_specific.local.papr_hpt.negative_test.not_ppc.required.non_ppc: PASS (2.02 s)
 (4/5) type_specific.local.papr_hpt.negative_test.not_ppc.enabled.non_ppc: PASS (2.02 s)
 (5/5) type_specific.local.papr_hpt.negative_test.not_ppc.disabled.non_ppc: PASS (2.00 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 14.41 s

File: libvirt/tests/cfg/papr_hpt.cfg [hunk 1]
```diff
@@ -0,0 +1,44 @@
+- papr_hpt:
+    type = papr_hpt
+    start_vm = no
+    status_error = False
+    variants:
+        - ppc_only:
+        - non_ppc:
+    variants:
+        - positive_test:
+            only ppc_only
```
Comment (chunfuwen) [inline libvirt/tests/cfg/papr_hpt.cfg]: only ppc_only duplicate meaning, suggest only ppc

File: libvirt/tests/cfg/papr_hpt.cfg [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            variants:
+                - disabled:
+                    resizing = disabled
+                    qemu_check = 'resize-hpt=disabled'
+                    error_msg = 'echo: write error: No such device'
+                - enabled:
+                    resizing = enabled
+                    qemu_check = 'resize-hpt=enabled'
+                    variants:
+                        - mem_hotplut:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/papr_hpt.cfg]: I assume mem_hotplug

Comment (chloerh) [reply libvirt/tests/cfg/papr_hpt.cfg]: Updated

File: libvirt/tests/cfg/papr_hpt.cfg (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/cfg/papr_hpt.cfg]: 'only ppc' could easily be affected by other arch configuration.

**Comment by dzhengfy** [comment]
@chloerh , it is better to paste your test result on Power8.

File: libvirt/tests/src/papr_hpt.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        vmxml.max_mem_rt_slots = int(params.get('max_mem_rt_slots', 16))
+        vmxml.max_mem_rt_unit = params.get('max_mem_rt_unit', 'KiB')
+
+        cpu = vm_xml.VMCPUXML()
+        cpu.xml = "<cpu><numa/></cpu>"
+
+        numa_cell = [eval(cell) for cell in params.get('numa_cell', '').split()]
+        logging.debug(numa_cell)
+
+        cpu.numa_cell = numa_cell
```
Comment (dzhengfy) [inline libvirt/tests/src/papr_hpt.py]: You configured 4 vcpu in your 'numa_cell' which requires your guest XML has 4 vcpu at least. 
If the vcpu specified in guest xml does not match your need, can you catch the error? Please have a test.

File: libvirt/tests/src/papr_hpt.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+from avocado.utils import process
+
+from virttest import virsh
+from virttest.libvirt_xml import vm_xml
+from virttest.libvirt_xml.devices.memory import Memory
+from virttest.utils_test import libvirt
+
+
+def run(test, params, env):
+    """
```
Comment (dzhengfy) [inline libvirt/tests/src/papr_hpt.py]: Input docstring

File: libvirt/tests/src/papr_hpt.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    resizing = params.get('resizing')
+    vm_name = params.get('main_vm')
+    vm = env.get_vm(vm_name)
+    status_error = eval(params.get('status_error', 'False'))
+    error_msg = params.get('error_msg', '')
+    hpt_order_path = params.get('hpt_order_path', '')
+    qemu_check = params.get('qemu_check', '')
+
+    def set_hpt(resizing, vmxml, sync=True):
```
Comment (dzhengfy) [inline libvirt/tests/src/papr_hpt.py]: docstring

File: libvirt/tests/src/papr_hpt.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        features_xml = vm_xml.VMFeaturesXML()
+        features_xml.hpt_resizing = resizing
+        vmxml.features = features_xml
+        if sync:
+            vmxml.sync()
+
+    def set_memory(vmxml):
```
Comment (dzhengfy) [inline libvirt/tests/src/papr_hpt.py]: docstring

File: libvirt/tests/src/papr_hpt.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        mem_xml.mem_model = params.get('mem_model', 'dimm')
+        return mem_xml
+
+    def check_qemu_cmd_line(content):
+        cmd = 'ps -ef|grep qemu'
+        qemu_line = process.run(cmd, shell=True).stdout
+        if content in qemu_line:
+            return True
+
+    def check_hpt_order(session, resizing=''):
```
Comment (dzhengfy) [inline libvirt/tests/src/papr_hpt.py]: docstring, please

File: libvirt/tests/cfg/papr_hpt.cfg (hunk 1 above)
Comment (balamuruhans) [reply libvirt/tests/cfg/papr_hpt.cfg]: `only ppc64,ppc64le` would this help ?

**Comment by balamuruhans** [comment]
@harihare would you like to review this PR ?

File: libvirt/tests/cfg/papr_hpt.cfg [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                            mem_model = dimm
+                            numa_cell = "{'id':'0','cpus':'0-1','memory':'524288','unit':'KiB'} {'id':'1','cpus':'2-3','memory':'524288','unit':'KiB'}"
+                            dmesg_content = Attempting to resize HPT to shift %d|HPT resize to shift %d complete
+
+        - negative_test:
+            status_error = True
+            error_msg = error: unsupported configuration: HPT tuning is only supported for pSeries guests
+            variants:
+                - not_ppc:
+                    only non_ppc
```
Comment (sathnaga) [inline libvirt/tests/cfg/papr_hpt.cfg]: `no ppc64,ppc64le` or `no pseries`

**Comment by iccaszhulili** [comment]
rh-caselink retest please

File: libvirt/tests/cfg/papr_hpt.cfg (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/cfg/papr_hpt.cfg]: updated

File: libvirt/tests/src/papr_hpt.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        new_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+        if 'ppc_only' in params['id']:
+            if arch.lower() != 'ppc64le':
+                test.cancel('This case is for PPC only')
+            set_hpt(resizing, new_xml)
+            if resizing == 'enabled':
+                set_memory(new_xml)
+
+            vm.start()
+            check_qemu_cmd_line(qemu_check)
```
[the remark this answers is not in the document]
Comment (chloerh) [reply libvirt/tests/src/papr_hpt.py]: updated

File: libvirt/tests/src/papr_hpt.py (hunk 5 above)
Comment (chloerh) [reply libvirt/tests/src/papr_hpt.py]: updated

**Comment by chloerh** [comment]
@chunfuwen @dzhengfy Could you review again, updated

**Pull request merged**